### DISCUSSION TITLE
assertJ와 junit의 차이점이 무엇일까?

### DISCUSSION THREAD

**Discussion by alstn113** [2024-woowacourse-study/level-interview#36] [Test]
.

**Comment by J-I-H-O** [comment]
JUnit은 단위 테스트를 위한 자바 프레임워크입니다. 
테스트 흐름을 제어하고, 테스트 수명 주기를 관리하는 등의 기능을 제공합니다.

AssertJ는 가독성 높은 테스트를 작성할 수 있는 라이브러리 입니다.
메서드 체이닝을 지원하여 테스트 코드를 보다 가독성 있고 표현력 있는 형태로 작성할 수 있도록 도와줍니다. 또한 좀 더 다양하고 풍부한 검증 기능을 제공합니다. 예를 들어 객체의 필드 값, 컬렉션의 크기, 예외 발생 여부 등을 검증할 수 있습니다.

**Reply by le2sky** [reply]
@J-I-H-O 
assertJ는 라이브러리라고 생각하고 있었는데, 땡이의 의견은 어떤가용?

**Reply by J-I-H-O** [reply]
생각치도 못한 부분이었네요. 관련 부분 더 조사해보았습니다.

- **JUnit**
  - 테스트 흐름 제어
    - `@Test` 어노테이션이 부착된 메서드를 찾아 실행
  - 테스트 수명 주기 관리
    - `@BeforeEach` 와 같은 어노테이션을 통해 테스트 케이스의 생명 주기를 관리
  - 기본적인 assert 메서드 제공
- **AssertJ**
  - 단순히 객체의 상태를 검증하기 위한 Assertion 기능 제공
    - JUnit 보다 풍부하고 가독성 높은 assert 메서드 제공
  - JUnit과 함께 사용

이렇게 비교해보았을 때 실행 흐름을 제어하는 JUnit은 프레임워크, 단순히 추가적인 검증 메서드를 제공하는 AssertJ는 라이브러리라고 말하는게 맞는 것 같습니다 👍

**Comment by 3Juhwan** [comment]
JUnit은 자바의 단위 테스트를 위한 프레임워크로, 주로 테스트 케이스를 정의하고 실행하는 데 사용됩니다. JUnit은 테스트 실행을 위한 어노테이션을 제공하고, 각 테스트 메서드의 성공 여부를 판단하고 보고하는 기능을 제공합니다.

반면에 AssertJ는 자바의 단언문을 강화하여 보다 표현력이 뛰어난 단언문을 사용할 수 있도록 해주는 라이브러리입니다. AssertJ는 JUnit과 함께 사용되어 기대되는 결과와 실제 결과를 비교하고 검증하는데 사용됩니다. AssertJ는 다양한 단언문을 제공하여 코드의 가독성을 향상시키고, 더 명확하고 표현력 있는 테스트 코드를 작성할 수 있도록 도와줍니다.

**Comment by GIVEN53** [comment]
Junit
테스트 흐름 제어
생명 주기 관리

assertJ
다양한 assert문
메서드 체이닝
가독성, 명확한 표현
